### PR TITLE
deprecate representation

### LINKED ISSUE #7032: Deprecate `representation` in frames and SkyCoord
More background on this in #6873.  There the old `representation` and `differential_cls` attributes/keyword arguments were shifted to `representation_type` and `differential_type`.  The stated plan there is to deprecate the "old" attributes in v3.1, and then switch to raising errors in 3.2.  This issue is for just the deprecation step.

### PR DESCRIPTION
**Pull request by eteq**
As detailed in #7032, this implements the plan to deprecate the "representation" keyword in coordinates.

Closes #7032 (although we should open a "remove it" issue for 3.2 presumably?)

### CODE REVIEW

File: astropy/coordinates/baseframe.py [hunk 1]
```diff
@@ -120,6 +120,13 @@ class for the representation of the base coordinates.  If a string,
     return repr_classes
 
 
+def _representation_deprecation():
+    """
+    Raises a deprecation warning for the "representation" keyword
+    """
+    warnings.warn('The `representation` keyword/property name is deprecated in '
+                  'favor of `representation_type`', AstropyDeprecationWarning)
```
[the remark this answers is not in the document]
Comment (eteq) [reply astropy/coordinates/baseframe.py]: Doh! Because I used `AstropyWarning` than quick fixed it in github without testing 🐑. Will fix.

**Comment by eteq** [comment]
Turns out there was a test failing due to this warning appearing somewhere another warning was being tested for.  I fixed it by being more fine-grained with the warning search. Hopefully this should be good now 🤞

File: CHANGES.rst [hunk 2]
```diff
@@ -968,6 +968,10 @@ Other Changes and Additions
 - Updated the bundled CFITSIO library to 3.450. See
   ``cextern/cfitsio/docs/changes.txt`` for additional information. [#8014]
 
+- The ``representation`` keywords in coordinate frames are now deprecated in
+  favor of the ``representation_type`` keywords (which are less
+  ambiguous). [#8119]
```
Comment (adrn) [inline CHANGES.rst]:
```suggestion
  ambiguously named). [#8119]
```

**Comment by adrn** [comment]
LGTM! One minor suggested wording change, but up to you whether to accept that change.

**Comment by eteq** [comment]
Thanks @adrn!

**Pull request merged**